lab-james/test/cowsay-controller-test.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+require('angular');
+require('angular-mocks');
+require('../app/entry.js');
+
+describe('testing cowsayController', function() {
+  beforeEach(() => {
+    angular.mock.module('cowsayApp');
+    angular.mock.inject(($rootScope, $controller) => {
+      this.$rootScope = $rootScope;
+      this.cowsayCtrl = new $controller('CowsayController');
+      this.cowsayCtrl.$onInit();
+    });
+  });
+
+  afterEach(() => this.$rootScope.$apply());
+
+  it('should have the correct starting state', () => {
+    expect(this.cowsayCtrl.update()).toEqual('moooo');
```

Comment:
So this test fails because, .update() returns a cowsay object not just a string. You can test for the .text value from the function if that is what you were trying to do